interfaces/cython/cantera/utils.pyx

```diff
@@ -117,5 +125,209 @@ cdef anyvalue_to_python(string name, CxxAnyValue& v):
+                type_j, ndim_j = get_types(j)
+                inner_types.add(type_j)
+                ndim_inner.add(ndim_j)
+            if len(inner_types) == 1 and len(ndim_inner) == 1:
+                return inner_types.pop(), ndim_inner.pop() + 1
+            else:
+                return None, 1
+        elif len(itype) == 1:
+            return itype.pop(), 1
+        else:
+            return None, 1
+
+cdef CxxAnyValue python_to_anyvalue(item, name=None) except *:
+    cdef CxxAnyValue v
+    if isinstance(item, dict):
+        v = dict_to_anymap(item)
+    elif isinstance(item, (list, tuple, set, np.ndarray)):
+        itype, ndim = get_types(item)
+        if ndim == 1:
+            if itype == str:
+                v = list_string_to_anyvalue(item)
+            elif itype == bool:
+                v = list_bool_to_anyvalue(item)
+            elif itype == numbers.Integral:
+                v = list_int_to_anyvalue(item)
+            elif itype == numbers.Real:
+                v = list_double_to_anyvalue(item)
+            elif itype == dict:
+                v = list_dict_to_anyvalue(item)
+            else:
+                v = list_to_anyvalue(item)
+        elif ndim == 2:
+            if itype == str:
+                v = list2_string_to_anyvalue(item)
+            elif itype == bool:
+                v = list2_bool_to_anyvalue(item)
+            elif itype == numbers.Integral:
+                v = list2_int_to_anyvalue(item)
+            elif itype == numbers.Real:
+                v = list2_double_to_anyvalue(item)
+            else:
+                v = list_to_anyvalue(item)
+        else:
+            v = list_to_anyvalue(item)
+    elif isinstance(item, str):
+        v = stringify(item)
+    elif isinstance(item, bool):
+        v = <cbool>(item)
+    elif isinstance(item, int):
+        v = <long int>(item)
+    elif isinstance(item, float):
+        v = <double>(item)
+    elif item is None:
+        pass  # None corresponds to "empty" AnyValue
+    elif name is not None:
+        raise CanteraError("Unable to convert item of type {!r}"
+            " with key {!r} to AnyValue".format(type(item), name))
+    else:
+        raise CanteraError("Unable to convert item of type {!r}"
+            " to AnyValue".format(type(item)))
+    return v
+
+# Helper functions for converting specific types to AnyValue
+
+cdef vector[CxxAnyValue] list_to_anyvalue(data) except *:
+    cdef vector[CxxAnyValue] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = python_to_anyvalue(item)
+    return v
+
+cdef vector[double] list_double_to_anyvalue(data):
+    cdef vector[double] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = <double>item
+    return v
+
+cdef vector[long] list_int_to_anyvalue(data):
+    cdef vector[long] v
+    v.resize(len(data))
+    cdef size_t i
+    for i, item in enumerate(data):
+        v[i] = <long>item
```

Comment:
If an `int` is passted to `python_to_anyvalue`, it gets type `long int`, but if a list of `int`s is passed, it ends up just being `long`. Is there a difference?